TypeError when GitHub URL is not defined #52

zubinbhasin opened the pull request:
#52

helio-frota commented:
@zubinbhasin thanks for the PR.
Can you please write a unit test for when Github URL is not found?
You can take a look here for similar examples https://github.com/nodeshift/npcheck/blob/main/test/plugins/archive.test.js#L80

aalykiot commented on src/plugins/archive.js:
```diff
@@ -1,8 +1,13 @@
 const archivePlugin = async (pkg, _, options) => {
+  if (!pkg.repository.url) {
+    warning(output.get());
+    return createWarning(`No Github url found for the ${pkg.name} module`);
```
```suggestion
    return createWarning(`The module "${pkg.name}" does not specify a GitHub URL.`);
```

zubinbhasin commented:
> @zubinbhasin thanks for the PR.
> Can you please write a unit test for when Github URL is not found?
> You can take a look here for similar examples https://github.com/nodeshift/npcheck/blob/main/test/plugins/archive.test.js#L80

I have added the unit test and made the changes according to the suggestions by @aalykiot

aalykiot commented on src/plugins/archive.js:
```diff
@@ -1,18 +1,24 @@
+    'Checking if github repository is archived'
+  ).withPadding(65);
+
+  if (!pkg.repository.url) {
```
If the `repository` field is also undefined then the if statement will throw a TypeError.

```suggestion
  if (!pkg?.repository?.url) {
```

aalykiot commented on test/plugins/archive.test.js:
```diff
@@ -95,3 +96,22 @@ it('should log failure message when module is archived', async () => {
+
+it('should log warning message when module does not specify its GitHub repository', async () => {
+  // mocking http request to GitHub
+  network.fetchGithub.mockImplementation(() => {
```
You don't have to mock the network request here since you'll never reach it.

aalykiot commented:
@zubinbhasin I would just writing one test with `const pkg = {};` and another with `const pkg = { repository: {} };` so we can cover all the edge cases 🙂

zubinbhasin commented:
> @zubinbhasin I would suggest writing one test with `const pkg = {};` and another with `const pkg = { repository: {} };` so we can cover all the edge cases 🙂

@aalykiot I've covered both of these cases, also removed the mock calls and added optional chaining snippet as mentioned above.

aalykiot reviewed (approved):
@zubinbhasin LGTM, can you please rebase from `main` so we can merge your PR ??

zubinbhasin commented:
> @zubinbhasin LGTM, can you please rebase from `main` so we can merge your PR ??

@aalykiot I've done the rebase but coveralls seems to be failing now.

helio-frota commented:
> @zubinbhasin LGTM, can you please rebase from main so we can merge your PR ??

> @aalykiot I've done the rebase but coveralls seems to be failing now.

Don't need to worry about that coveralls issue :+1:
